src/dataflow/src/source/s3.rs

```diff
@@ -214,8 +218,15 @@ async fn download_objects_task(
                     seen_buckets.insert(msg.bucket.clone(), bi);
                 };
 
-                let update =
-                    download_object(&tx, &activator, &client, msg.bucket.clone(), msg.key).await;
+                let update = download_object(
+                    &tx,
+                    &activator,
+                    &client,
+                    msg.bucket.clone(),
+                    msg.key,
+                    compression.clone(),
```

Comment:
it's probably worth just adding `Copy` to the list of derives on dataflow-types::Compression instead of cloning it everywhere.